src/main/kotlin/controllers/PersonController.kt

```diff
@@ -38,40 +38,34 @@ class PersonController(
 
     @GetMapping
     @PreAuthorize("hasAuthority('PersonAuthority.ADMIN')")
-    fun findAll(pageable: Pageable, principal: Principal) = principal
-        .findUser()
-        ?.let {
-            service
-                .findAll(pageable)
-                .toResponse()
-        }
-        ?: throw ResponseStatusException(UNAUTHORIZED)
+    fun findAll(pageable: Pageable, principal: Principal) = service
+        .findAll(pageable)
+        .toResponse()
 
     @GetMapping("/{code}")
     @PreAuthorize("hasAuthority('PersonAuthority.READ')")
     fun findByCode(@PathVariable code: String, principal: Principal): ResponseEntity<Person> = principal
         .findUser()
         ?.let {
-            return@let when {
+            when {
                 it.isAdmin() -> service.findByCode(code)?.toResponse()
                 else -> service.findByUserCode(it.code)?.toResponse()
             }
-            ?: throw ResponseStatusException(NOT_FOUND, "No Item found with this PersonCode")
+                ?: throw ResponseStatusException(NOT_FOUND, "No Item found with this PersonCode")
         }
         ?: throw ResponseStatusException(UNAUTHORIZED)
 
     @PostMapping
     @PreAuthorize("hasAuthority('PersonAuthority.WRITE')")
     fun post(@RequestBody form: PersonForm, principal: Principal) = principal
         .findUser()
-        ?.let {
+        ?.let { user ->
             val userCode = when {
-                it.isAdmin() -> form.userCode
-                else -> it.code
+                user.isAdmin() -> null
+                else -> user.code
             }
             form.copy(userCode = userCode)
-
-            return@let service.create(form)
+                .let { service.create(it) }
```

Comment:
is there an advantage doing it this way or just preference? 
Im very much in favor of consistency.
(my personal dislike of curly brackets and thus favoring `return@let`)

Reply:
I'm not sure if both give the same output. My preference is to write in a more functional style. I will check the return@let syntax I have never seen it before.

Reply:
dont look too far ;) https://kotlinlang.org/docs/reference/returns.html